### PR TITLE
:book:  doc/development: add devstack notes

### PR DESCRIPTION
**Pull request by tormath1**
this is notes from my setup, it can be useful for folks who want to develop CAPO with a single DevStack instance.
<hr>

**What this PR does / why we need it**: This PR adds some documentation notes for CAPO development on a single DevStack instance.

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: docs/book/src/development/development.md [hunk 1]
```diff
@@ -195,6 +196,113 @@ and create the environment by running:
...
+
+# /etc/glance/glance-image-import.conf
+[image_import_opts]
+image_import_plugins = ['image_decompression']
+
+$ sudo systemctl restart devstack@g-api.service apache2
+```
+
+With this dev setup, it might be useful to enable DHCP for the public subnet:
+Admin > Network > Networks > `public` > Subnets > `public-subnet` > Edit Subnet > Subnet Details > :ballot_box_with_check: Enable DHCP + Add DNS
```
Comment (dulek) [inline docs/book/src/development/development.md:261]: Where's that done?

File: docs/book/src/development/development.md [hunk 2]
```diff
@@ -195,6 +196,113 @@ and create the environment by running:
...
+DATABASE_PASSWORD=\$ADMIN_PASSWORD
+RABBIT_PASSWORD=\$ADMIN_PASSWORD
+SERVICE_PASSWORD=\$ADMIN_PASSWORD
+
+GIT_BASE=https://opendev.org
+# Enable Logging
+LOGFILE=$DEST/logs/stack.sh.log
+VERBOSE=True
+LOG_COLOR=True
+enable_service rabbit
```
Comment (dulek) [inline docs/book/src/development/development.md:226]: Not needed, it's in the defaults: https://github.com/openstack/devstack/blob/c424a7a299e37004d318107648bb18e157344985/stackrc#L64-L83

File: docs/book/src/development/development.md [hunk 3]
```diff
@@ -195,6 +196,113 @@ and create the environment by running:
...
+enable_plugin neutron $GIT_BASE/openstack/neutron
+# Octavia supports using QoS policies on the VIP port:
+enable_service q-qos
+enable_service placement-api placement-client
```
Comment (dulek) [inline docs/book/src/development/development.md:230]: Same here.

**Review by dulek — CHANGES_REQUESTED** [review]

File: docs/book/src/development/development.md (hunk 1 above)
Comment (dulek) [reply docs/book/src/development/development.md:259]: Worth to say that. But in general I'd prefer doing that from the CLI instead. I'm guessing it's `enable_dhcp` option on the subnet?

File: docs/book/src/development/development.md [hunk 4]
```diff
@@ -195,6 +196,113 @@ and create the environment by running:
...
+# Octavia services
+enable_plugin octavia $GIT_BASE/openstack/octavia master
+enable_plugin octavia-dashboard $GIT_BASE/openstack/octavia-dashboard
+enable_plugin ovn-octavia-provider $GIT_BASE/openstack/ovn-octavia-provider
+enable_plugin octavia-tempest-plugin $GIT_BASE/openstack/octavia-tempest-plugin
+enable_service octavia o-api o-cw o-hm o-hk o-da
+# Cinder
+enable_service c-api c-vol c-sch
```
[the remark this answers is not in the document]
Comment (dulek) [reply docs/book/src/development/development.md:238]: Hm, maybe one of the plugins disable that? Anyway we get e.g. n-cpu somehow, so I guess it's somewhere else.

**Comment by tobiasgiese** [comment]
Do you know my WIP PR https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1539? I think this is mainly the same as your documentation

**Review by mdbooth — APPROVED** [review]

**Review by lentzi90 — COMMENTED** [review]
This has been hanging for a while. I would say it is better to have the notes merged than polishing them forever so merging.
/lgtm
/hold cancel

**Pull request merged**